### PR TITLE
only include media posts by post creator

### PR DESCRIPTION
**Pull request by estrattonbailey**
Fixes https://github.com/bluesky-social/social-app/issues/1167

### CODE REVIEW

File: packages/bsky/tests/views/author-feed.test.ts [hunk 1]
```diff
@@ -236,51 +236,44 @@ describe('pds author feed views', () => {
...
-            isImageEmbed(post.embed?.media)) ||
-          isImageEmbed(post.embed)
-        )
+      carolFeed.feed.every(({ post }) => {
+        const isRecordWithActorMedia =
+          isEmbedRecordWithMedia(post.embed) && isImageEmbed(post.embed?.media)
+        const isActorMedia = isImageEmbed(post.embed)
+        const isFromActor = post.author.did === carol
+
+        return (isRecordWithActorMedia || isActorMedia) && isFromActor
```
Comment (estrattonbailey) [inline packages/bsky/tests/views/author-feed.test.ts:253]: Tightened up these tests to ensure that the author is the actor in question

File: packages/bsky/tests/views/author-feed.test.ts [hunk 2]
```diff
@@ -236,51 +236,44 @@ describe('pds author feed views', () => {
...
 
+    const { data: danFeed } = await agent.api.app.bsky.feed.getAuthorFeed({
+      actor: dan,
+      filter: 'posts_with_media',
+    })
+
+    expect(danFeed.feed.length).toEqual(0)
+  })
+
+  it('filters by posts_no_replies', async () => {
```
Comment (estrattonbailey) [inline packages/bsky/tests/views/author-feed.test.ts:276]: And split a test out

File: packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts [hunk 3]
```diff
@@ -38,17 +39,37 @@ export default function (server: Server, ctx: AppContext) {
...
+      } else {
+        const actorRes = await ctx.db.db
+          .selectFrom('did_handle')
+          .select('did')
+          .where('did_handle.handle', '=', actor)
+          .executeTakeFirst()
+        if (actorRes) {
+          actorDid = actorRes?.did
+        }
+      }
```
Comment (estrattonbailey) [inline packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts:55]: Moved this out of `getFeedItemsQb` using the same approach as the App View

File: packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts [hunk 4]
```diff
@@ -38,17 +39,37 @@ export default function (server: Server, ctx: AppContext) {
       const feedService = ctx.services.appView.feed(ctx.db)
       const graphService = ctx.services.appView.graph(ctx.db)
 
+      // maybe resolve did first
```
Comment (dholms) [inline packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts:42]: there's a method on actorService for this that may be nice to use: `getActorDid`

**Review by dholms — APPROVED** [review]
looks great 👍 

think you can use the existing `getActorDid` method to simplify it a bit (I imagine we're going to start sprinkling that around a bit more)

File: packages/bsky/tests/views/author-feed.test.ts [hunk 5]
```diff
@@ -165,12 +165,14 @@ describe('pds author feed views', () => {
...
       )
 
-    const { data: postBlock } = await agent.api.app.bsky.feed.getAuthorFeed(
-      { actor: alice },
-      { headers: await network.serviceHeaders(carol) },
-    )
-
-    expect(postBlock.feed.length).toEqual(0)
+    try {
+      const { data: postBlock } = await agent.api.app.bsky.feed.getAuthorFeed(
```
Comment (dholms) [inline packages/bsky/tests/views/author-feed.test.ts:169]: this test actually wont fail if it doesn't throw

normally use the following syntax
```ts
await expect(function).rejects.toThrow("Profile not found")
```

**Review by dholms — APPROVED** [review]

**Comment by pfrazee** [comment]
Sweet!

**Pull request merged**